### PR TITLE
Create missing index for the table IDN_OAUTH2_TOKEN_BINDING

### PR DESCRIPTION
**Pull request by UdeshAthukorala**
## Purpose
Create missing index for the table IDN_OAUTH2_TOKEN_BINDING.

Related PR : https://github.com/wso2/carbon-identity-framework/pull/3527
Related Issue : https://github.com/wso2/product-is/issues/11610

### CODE REVIEW

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/oracle_rac.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CREATE OR REPLACE PROCEDURE add_index_if_not_exists (query IN VARCHAR2)
+  IS
+BEGIN
+  execute immediate query;
+  dbms_output.put_line(query);
+exception WHEN OTHERS THEN
+  dbms_output.put_line( 'Skipped ');
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/oracle_rac.sql:7]:
```suggestion
  dbms_output.put_line( 'Skipped');
```

Comment (UdeshAthukorala) [reply components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/oracle_rac.sql]: Addressed with https://github.com/wso2-extensions/identity-migration-resources/pull/194/commits/abd3835cf84f288e4cfb37ab7f2e725cdd364b71

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+CREATE OR REPLACE FUNCTION skip_index_if_exists(nameOfTheIndex varchar(64),tableName varchar(64), tableColumns varchar(64)) RETURNS void AS $$ declare s varchar(1000); declare result INTEGER; begin SELECT COUNT(1) into result FROM pg_indexes WHERE indexname = nameOfTheIndex; IF result = 0 THEN s :=  CONCAT('CREATE INDEX ' , nameOfTheIndex , ' ON ' , tableName, tableColumns); execute s; end if; END;$$ LANGUAGE plpgsql;
+
+SELECT skip_index_if_exists('idx_tk_value_type','idn_oauth2_token_binding','(token_binding_value, token_binding_type)');
+
+DROP FUNCTION skip_index_if_exists(varchar,varchar,varchar);
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql:5]:
```suggestion
DROP FUNCTION skip_index_if_exists(varchar,varchar,varchar);

```

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+CREATE OR REPLACE FUNCTION skip_index_if_exists(nameOfTheIndex varchar(64),tableName varchar(64), tableColumns varchar(64)) RETURNS void AS $$ declare s varchar(1000); declare result INTEGER; begin SELECT COUNT(1) into result FROM pg_indexes WHERE indexname = nameOfTheIndex; IF result = 0 THEN s :=  CONCAT('CREATE INDEX ' , nameOfTheIndex , ' ON ' , tableName, tableColumns); execute s; end if; END;$$ LANGUAGE plpgsql;
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql:1]:
```suggestion
CREATE OR REPLACE FUNCTION skip_index_if_exists(nameOfTheIndex varchar(64), tableName varchar(64), tableColumns varchar(64)) RETURNS void AS $$ declare s varchar(1000); declare result INTEGER; begin SELECT COUNT(1) into result FROM pg_indexes WHERE indexname = nameOfTheIndex; IF result = 0 THEN s :=  CONCAT('CREATE INDEX ' , nameOfTheIndex , ' ON ' , tableName, tableColumns); execute s; end if; END;$$ LANGUAGE plpgsql;
```

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql (hunk 2 above)
Comment (UdeshAthukorala) [reply components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql]: Addressed with https://github.com/wso2-extensions/identity-migration-resources/pull/194/commits/f3ba60c9f39ca00da11be60a646fcacf811ce9ce

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql (hunk 3 above)
Comment (UdeshAthukorala) [reply components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql]: Addressed with https://github.com/wso2-extensions/identity-migration-resources/pull/194/commits/f3ba60c9f39ca00da11be60a646fcacf811ce9ce

**Comment by Buddhimah** [comment]
@UdeshAthukorala Please squash the commits into one commit

**Review by Buddhimah — APPROVED** [review]

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+CREATE OR REPLACE FUNCTION skip_index_if_exists(nameOfTheIndex varchar(64), tableName varchar(64), tableColumns varchar(64)) RETURNS void AS $$ declare s varchar(1000); declare result INTEGER; begin SELECT COUNT(1) into result FROM pg_indexes WHERE indexname = nameOfTheIndex; IF result = 0 THEN s :=  CONCAT('CREATE INDEX ' , nameOfTheIndex , ' ON ' , tableName, tableColumns); execute s; end if; END;$$ LANGUAGE plpgsql;
```
Comment (emswbandara) [inline components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql:1]: this exceeds 120 character length limit right?
Shall we fix it

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/mysql.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+DROP PROCEDURE IF EXISTS skip_index_if_exists;
+
+CREATE PROCEDURE skip_index_if_exists(indexName varchar(64), tableName varchar(64), tableColumns varchar(255)) BEGIN BEGIN DECLARE CONTINUE HANDLER FOR SQLEXCEPTION BEGIN END; SET @s = CONCAT('CREATE INDEX ', indexName, ' ON ', tableName, tableColumns); PREPARE stmt FROM @s; EXECUTE stmt; END; END;
```
Comment (emswbandara) [inline components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/mysql.sql:3]: this exceeds 120 character length limit right?
Shall we fix it

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql (hunk 4 above)
Comment (UdeshAthukorala) [reply components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/postgresql.sql:1]: This file was modified in the main branch.

File: components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/mysql.sql (hunk 5 above)
Comment (UdeshAthukorala) [reply components/org.wso2.is.migration/migration-resources/5.12.0/dbscripts/step6/identity/mysql.sql:3]: This file was modified in the main branch.

**Pull request merged**